### PR TITLE
fix: channel deletion mapping bug

### PR DESCRIPTION
**Pull request by zeeshanakram3**
addresses #4601

### CODE REVIEW

File: query-node/mappings/src/content/channel.ts [hunk 1]
```diff
@@ -412,3 +420,31 @@ async function processModeratorRemark(
...
+    return await store.getMany(entityType, {
+      where: { channel: { id: channelId } } as FindOptionsWhere<T>,
+    })
+  }
+
+  const removeRelations = async <T>(store: DatabaseManager, entities: T[]) => {
+    await Promise.all(entities.map(async (r) => await store.remove<T>(r)))
+  }
+
+  const referencingEntities: typeof BaseModel[] = [Collaborator, ChannelNftCollectors, MemberBannedFromChannelEvent]
```
Comment (Lezek123) [inline query-node/mappings/src/content/channel.ts]: If you declare it as:
```typescript
  const referencingEntities: { new (): BaseModel & { channel: Partial<Channel> } }[] = [
    Collaborator,
    ChannelNftCollectors,
    MemberBannedFromChannelEvent,
  ]
```
then you can remove `entity as any` assertion below and preserve the typesafety

**Review by Lezek123 — CHANGES_REQUESTED** [review]
Additional notes:
1. I don't think we have any tests for banning members from a channel. If it's not time-consuming, could you add a call to `BanOrUnbanMemberFixture` in `flows/commentsAndReactions`, to make sure removing `MemberBannedFromChannelEvent` entities works as expected?

File: query-node/mappings/src/content/channel.ts (hunk 1 above)
Comment (zeeshanakram3) [reply query-node/mappings/src/content/channel.ts]: Adressed in [550314a](https://github.com/Joystream/joystream/pull/4603/commits/550314a18c8fe98af2eb5b65e96815cda80bb394)

**Review by Lezek123 — APPROVED** [review]
LGTM, but will wait until the checks pass before merging.

A note about updating atlas-dev: Because the atlas-dev network is running from `carthage` branch and the mappings between `carthage` and `master` are not fully compatible (there was `Gateway` => `Apps` working group renaming), you probably just want to cherry-pick [463c141](https://github.com/Joystream/joystream/pull/4603/commits/463c141bdc9b69988b3916f943801e003cab3e6b) and [550314a](https://github.com/Joystream/joystream/pull/4603/commits/550314a18c8fe98af2eb5b65e96815cda80bb394) into `carthage` before re-building and restarting the processor.

**Pull request merged**